src/data_source/storage/sql.rs

```diff
@@ -361,10 +361,29 @@ impl Config {
 pub struct SqlStorage {
     client: Arc<Client>,
     tx_in_progress: bool,
-    pruner_cfg: Option<PrunerCfg>,
+    pruner: Pruner,
     _connection: BackgroundTask,
 }
 
+#[derive(Debug, Default)]
+pub struct Pruner {
+    enabled: bool,
+    config: PrunerCfg,
+    pruned_height: u64,
+    run_in_progress: bool,
+}
+
+impl Pruner {
+    fn new(config: PrunerCfg) -> Self {
+        Pruner {
+            enabled: true,
+            config,
+            pruned_height: 0,
```

Comment:
Why does `pruned_height` start at 0? Shouldn't it be loaded from the db?

Reply:
Oh, I see, is it because this is only used when `run_in_progress`?

Reply:
removed run_in_progress field and made pruned_height optional to make it more clear